### PR TITLE
Feature 13412 - Allow Changing Completing Party in Registrations

### PR DESCRIPTION
**Pull request by ritvick**
*Issue #:* /bcgov/entity/issues/13412

*Description of changes:*
We want to keep the orgperson detail emplty in case of registration by a staff. So that staff will manually have to fill the information on behalf of the customer.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Review by severinbeauvais — COMMENTED** [review]
This looks fine but does it fully satisfy the ticket requirements?

Also, please update the app version and commit the package files.

**Comment by ritvick** [comment]
> This looks fine but does it fully satisfy the ticket requirements?
> 
> Also, please update the app version and commit the package files.

Yeah it fulfils all the requirement, unless I am missing something? Can you be more specific what equirements I may have missed here?

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "business-create-ui",
-  "version": "4.2.0",
+  "version": "4.2.1",
```
Comment (severinbeauvais) [inline package.json:3]: This project is [already at v4.2.1](https://github.com/bcgov/business-create-ui/blob/c1afbed036ccd93f75b7da4be7994e3ff79c2647/package.json#L3). Please rebase and then update app version again.

**Comment by severinbeauvais** [comment]
> Yeah it fulfils all the requirement, unless I am missing something? Can you be more specific what equirements I may have missed here?

When I refer back to the ticket, I see that Completing Party should be blank and editable for SBC Staff (which is not the same as BCROS staff). I think this isn't done.

**Comment by severinbeauvais** [comment]
I have added some reviewers to this ticket. I cannot be the sole reviewer for Entity UIs. From now on, I will wait for other reviewers to have a look at PRs before I do.

cc: @pwei1018 @davemck513

File: src/components/Registration/RegPeopleAndRoles.vue [hunk 2]
```diff
@@ -223,8 +223,8 @@ export default class RegPeopleAndRoles extends Mixins(PeopleRolesMixin) {
     // assign party type (org or person)
     this.currentOrgPerson.officer.partyType = partyType
 
-    // pre-populate Completing Party's name, email address and mailing address
-    if (roleType === RoleTypes.COMPLETING_PARTY && partyType === PartyTypes.PERSON) {
+    // pre-populate Completing Party's name, email address and mailing address only if the logged in user is not staff
+    if (roleType === RoleTypes.COMPLETING_PARTY && partyType === PartyTypes.PERSON && !this.isRoleStaff) {
```
Comment (lambert-alex) [inline src/components/Registration/RegPeopleAndRoles.vue:227]: I think we should have a unit test checking this new functionality.

Comment (ritvick) [reply src/components/Registration/RegPeopleAndRoles.vue]: Hi @lambert-alex ,
I am mostly backend side and finiding it difficult to how to put unit test for this. 
I checked the unit tests folder but was not able to locate tests which are related to logged in user behaviour. 
Can you point to some if they already exists ( Probably I would not have looked in detail) ?

Comment (severinbeauvais) [reply src/components/Registration/RegPeopleAndRoles.vue]: In my opinion, the best way to handle this (and the lack of tests elsewhere) is to create a "staff" test suite where we can mount different components and verify that they display/hide the expected elements.

As this ^^ is beyond the scope of this ticket, Ritvick, please create a Tech Debt ticket (in Entities team space) to do this.

**Review by severinbeauvais — APPROVED** [review]
LGTM.

Please update your ticket with the current Sprint and Releases. Also please add a Test Notes comment identifying what to test (for QA purposes). Thanks!

**Review by lewischenstudio — APPROVED** [review]
LGTM 👍 It would be nice to add a test case for SbcStaff or add a test note for QA.

File: src/components/Registration/RegPeopleAndRoles.vue (hunk 2 above)
Comment (ritvick) [reply src/components/Registration/RegPeopleAndRoles.vue]: Created Tech debt ticket here: https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/bcgov/entity/13614

**Comment by ritvick** [comment]
Added test Notes in the original ticket.

**Comment by lambert-alex** [comment]
> Added test Notes in the original ticket.

Thank you!

File: src/components/Registration/RegPeopleAndRoles.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/Registration/RegPeopleAndRoles.vue]: @lambert-alex OK to resolve this conversation?

**Pull request merged**